### PR TITLE
feat(node): extract overrides from Config into Settings

### PR DESCRIPTION
**Pull request by Wondertan**
## About 
#290 adds Component overriding options to `node` and applies those onto Config. This is confusing and this PRs aims to fix this by introducing `node.settings`.

### CODE REVIEW

File: node/init.go [hunk 1]
```diff
@@ -13,10 +13,10 @@ import (
 // Init initializes the Node FileSystem Repository for the given Node Type 'tp' in the directory under 'path' with
 // default Config. Options are applied over default Config and persisted on disk.
 func Init(path string, tp Type, options ...Option) error {
-	cfg := DefaultConfig(tp)
+	cfg, sets := DefaultConfig(tp), new(settings)
```
Comment (adlerjohn) [inline node/init.go:16]: Not the biggest fan of `sets`. Is there a better name? Maybe even `s`.

File: node/fxutil/options.go [hunk 2]
```diff
@@ -131,6 +131,11 @@ func OverrideSupply(vals ...interface{}) Option {
 				refVal, tp = refVal.Elem(), tp.Elem()
 			}
 
+			if !refVal.IsValid() {
+				// if there was nil - ignore
+				continue
```
Comment (adlerjohn) [inline node/fxutil/options.go:136]: Is it okay to silently ignore an invalid thing here?

File: node/init.go (hunk 1 above)
Comment (Wondertan) [reply node/init.go:16]: Don't have strong preferences about naming here. Btw, why?

File: node/fxutil/options.go (hunk 2 above)
Comment (Wondertan) [reply node/fxutil/options.go:136]: Good question. All the __really__ invalid cases are checked and errors for them are thrown above the line. Regarding this case, it is basically `nil` and it is usually fine to allow it to be passed in the optional pattern, e.g  `WithSomething(nil)`, so you don't need to check for `if something != nil { WithSomething(something) }`.

File: node/init.go (hunk 1 above)
Comment (adlerjohn) [reply node/init.go:16]: "sets" makes me initially think of https://en.wikipedia.org/wiki/Set_(mathematics)

File: node/fxutil/options.go (hunk 2 above)
Comment (adlerjohn) [reply node/fxutil/options.go:136]: Ah okay. Can you document this in a comment?

**Review by renaynay — APPROVED** [review]
I'm good with this as long as @adlerjohn 's comments are addressed.

**Review by renaynay — APPROVED** [review]

**Review by Bidon15 — APPROVED** [review]

**Pull request merged**